src/main/scala/org/constellation/util/Metrics.scala

```diff
@@ -228,7 +228,7 @@ class Metrics(periodSeconds: Int = 1)(implicit dao: DAO)
       updateMetric("channelCount", dao.threadSafeMessageMemPool.activeChannels.size)
 
       // Get TransactionServiceDBSize
-      val txServiceMap = dao.transactionService.getMetricsMap
+      val txServiceMap = dao.transactionService.getMetricsMap.unsafeRunSync()
```

Comment:
you could use updateMetricAsync and chain IO's

Reply:
:+1: